dev-tools/reconfigurator-cli/src/lib.rs

```diff
@@ -837,6 +841,44 @@ struct LoadExampleArgs {
     /// Do not create entries for disks in the blueprint.
     #[clap(long)]
     no_disks_in_blueprint: bool,
+
+    /// Set a 0-indexed sled's policy
+    #[clap(long, value_name = "INDEX:POLICY")]
+    sled_policy: Vec<LoadExampleSledPolicy>,
+}
+
+#[derive(Clone, Debug)]
+struct LoadExampleSledPolicy {
+    /// The index of the sled to set the policy for.
+    index: usize,
+
+    /// The policy to set.
+    policy: SledPolicy,
+}
+
+impl FromStr for LoadExampleSledPolicy {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let Some((index, policy)) = s.split_once(':') else {
+            return Err(anyhow!("invalid format, expected <index>:<policy>"));
+        };
+        let index = index.parse().with_context(|| {
+            format!("error parsing sled index `{index}` as a usize")
+        })?;
+        let policy = SledPolicyOpt::from_str(
+            policy, /* ignore_case */ false,
+        )
+        .map_err(|_message| {
+            // _message is just something like "invalid variant: <value>".
+            // We choose to use our own message instead.
```

Comment:
Can we change the error message on `SledPolicyOpt` itself so all its users get this nicer message?

Reply:
Ah unfortunately `SledPolicyOpt` is generated by the `ValueEnum` proc macro. The error with clap is reasonable:

![image](https://github.com/user-attachments/assets/28b9043f-1593-4b6d-9748-f78d264234f6)

With `load-example --sled-policy` we get a different error message, but one that still conveys the same information:

![image](https://github.com/user-attachments/assets/cca5e3c5-6048-48a6-86d4-f597898446fe)